src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/DotNetCoreProjectCompatibilityDetector.cs

```diff
@@ -0,0 +1,317 @@
+                        if(suppressPrompt && settingsManager != null)
+                        {
+                            await settingsManager.SetValueAsync(SuppressDotNewCoreWarningKey, true, isMachineLocal: true).ConfigureAwait(true);
+                        }
+                    }
+                }
+                else
+                {
+                    DialogServices.Value.DontShowAgainMessageBox(caption, string.Format(VSResources.NotSupportedDotNetCoreProject, s_partialSupportedVersion.Major, s_partialSupportedVersion.Minor), 
+                                                                 null, false, VSResources.LearnMore, LearnMoreFwlink);
+                }
+            }
+        }
+
+        private async Task<CompatibilityLevel> GetProjectCompatibilityAsync(UnconfiguredProject project)
+        {
+            if (project.Capabilities.AppliesTo(ProjectCapability.CSharpOrVisualBasicOrFSharp))
+            {
+                IProjectProperties properties = project.Services.ActiveConfiguredProjectProvider.ActiveConfiguredProject.Services.ProjectPropertiesProvider.GetCommonProperties();
+                string tfm = await properties.GetEvaluatedPropertyValueAsync("TargetFrameworkMoniker").ConfigureAwait(false);
+                var fw = new FrameworkName(tfm);
+                if ((fw.Identifier.Equals(".NETCore", StringComparison.OrdinalIgnoreCase) || fw.Identifier.Equals(".NETCoreApp", StringComparison.OrdinalIgnoreCase)))
+                {
+                    if(fw.Version.Major == s_partialSupportedVersion.Major && fw.Version.Minor == s_partialSupportedVersion.Minor)
+                    {
+                        return CompatibilityLevel.Partial;
+                    }
+                    else if(fw.Version.Major > s_partialSupportedVersion.Major || (fw.Version.Major == s_partialSupportedVersion.Major && fw.Version.Minor > s_partialSupportedVersion.Minor))
+                    {
+                        return CompatibilityLevel.NotSupported;
+                    }
+                }
+            }
+
+            return CompatibilityLevel.Supported;
+        }
+
+        /// <summary>
+        /// Query the ChannelSuffix property. Will be empty if this is the release channel
+        /// </summary>
+        private bool IsReleaseChannel()
+        {
+            IVsAppId vsAppId = _serviceProvider.GetService<IVsAppId, SVsAppId>();
+            int hr = vsAppId.GetProperty((int)VSAPropID.VSAPROPID_ChannelSuffix, out object value);
+            return ErrorHandler.Succeeded(hr) && value is string && string.IsNullOrEmpty((string)value);
+        }
+
+        #region Unused
+        
+        /// <summary>
+        /// Unused IVsSolutionEvents
+        /// </summary>
+        public int OnAfterOpenSolution(object pUnkReserved, int fNewSolution)
+        {
+            return VSConstants.S_OK;
```

Comment:
Is it better to return E_NOTIMPL to avoid repeated calls?  (Not sure if IVsSolutionEvents does that trick or not).

Reply:
I don't know if it will handle E_NOTIMPL - it is possible it treats it as an error. I'd have to check the sources. I think S_OK is find here since this is just a notification.